pkg/config/autodiscovery/autodiscovery.go

```diff
@@ -82,37 +81,25 @@ func DiscoverComponentsFromEnv() ([]config.ConfigurationProviders, []config.List
 		return detectedProviders, detectedListeners
 	}
 
-	// Upon retiring this flag, see comment in `KubeContainerConfigProvider`
```

Comment:
Referring to [this comment](https://github.com/DataDog/datadog-agent/blob/main/pkg/autodiscovery/providers/kubecontainer.go#L26-L30) 

Basically there are now 2 unused `Provider`s. This cleanup could be deferred to the @DataDog/container-integrations team as [they own the providers](https://github.com/DataDog/datadog-agent/blob/main/.github/CODEOWNERS#L203), but if we don't do it in this PR lets make sure we track that work.

Reply:
That's a good question. @juliogreff - would you like me to perform this cleanup now, or do you plan to work on this code when refactoring AD? (Just asking so we don't do redundant work)

Reply:
I'm not gonna get to the AD refactoring so soon, and as far as providers are concerned, the refactoring is already done in the form of the `KubeContainerConfigProvider`, the remaining work is indeed just removing the older ones. Feel free to remove them already 🚀